src/Marten/Schema/Identity/Sequences/HiLoSequence.cs

```diff
@@ -91,14 +81,26 @@ public void AdvanceToNextHi()
 
                 try
                 {
-                    var tx = conn.BeginTransaction(IsolationLevel.Serializable);
-                    var raw = conn.CreateCommand().CallsSproc(GetNextFunction)
-                        .With("entity", _entityName)
-                        .Returns("next", NpgsqlDbType.Bigint).ExecuteScalar();
+                    var attempts = 0;
+                    do
+                    {
+                        attempts++;
+
+                        // Sproc is expected to return -1 if it's unable to
+                        // atomically secure the next hi
+                        var raw = conn.CreateCommand().CallsSproc(GetNextFunction)
+                            .With("entity", _entityName)
+                            .Returns("next", NpgsqlDbType.Bigint).ExecuteScalar();
+
+                        CurrentHi = Convert.ToInt64(raw);
 
-                    tx.Commit();
+                    } while (CurrentHi < 0 && attempts < 30);
```

Comment:
I think it would help to pass the `attempts` value as in `StoreOptions` with a default value of `30` and use it so that end users can configure it as they see it fit.

Also I think adding some sleep time between attempts would help.

Reply:
Added new property under `StoreOptions.HiloSequenceDefaults` named `MaxAdvanceToNextHiAttempts`.

No point in adding sleep time. You would want to retry as soon as possible.